src/sb-util.ts

```diff
@@ -49,12 +39,26 @@ export class ScratchProject {
 		return new SpriteCollection(sprites).query(selector);
 	}
 
-	stage() {
+	stage(): Sprite {
 		return this.sprites('[isStage=true]').first();
 	}
 
 	blocks() {
-		throw new Error('ScratchProject blocks() function not implemented yet!');
+		const sprites = this.sprites();
+		const iters = [];
+		for(const sprite of sprites) {
+			const spriteBlocks: BlockCollection = sprite.blocks();
+
+			// Iterating over a BlockCollection will return a Block as each element.
+			// Therefore, we should convert the Block to a raw BlockProperties object
+			//	which can be obtained by accessing the Block in storage and retrieving
+			//	the first element
+			const spriteBlocksProps = makeIterable(spriteBlocks, (s: Block) => map(s, s => storage.get(s).slice().pop()));
```

Comment:
this might be the first argument for `Block.props()` to get the BlockProperties I've seen

Reply:
I think I'd also rather see this exposed from the Collection as a secondary iterator... `BlockCollection.rawIteraror()` or something?

Reply:
Cool! I see why we would need a props() function.

Reply:
Implement a `last` helper that returns the last member instead of `.slice().pop()`.

Reply:
I like the rawIterator. It be nice to provide at least a private facility to avoid unnecessary Sprite, Block, etc instantiation and that way the iterators only instantiate values that the user will use. That could save on a lot of processing and memory.